iOS_SDK/OneSignalSDK/Source/OneSignalWebView.m

```diff
@@ -133,7 +131,15 @@ - (void)showInApp {
     @catch(NSException* exception) { }
 }
 
+- (void)clearWebView {
+    [_webView loadHTMLString:@"" baseURL:nil];
+    if (viewControllerForPresentation)
+        [viewControllerForPresentation.view removeFromSuperview];
+}
 
+- (void)presentationControllerDidDismiss:(UIPresentationController *)presentationController {
+    [self clearWebView];
+}
```

Comment:
What is the thought and intention with implementing `UIAdaptivePresentationControllerDelegate`?

Are both methods required (`dismiss` and `presentationControllerDidDismiss`) or is one better to go with than the other? 

Or do they trigger in different scenarios and this solves an edge case?

Reply:
> What is the thought and intention with implementing UIAdaptivePresentationControllerDelegate?

The default `modalPresentationStyle` in iOS 13 is usually `pageSheet`, which can be dismissed by swiping down. Current implementation assumes that user can only dismiss the viewcontroller by tapping `leftBarButtonItem`, and necessary clean-up is done in `dismiss` method. When user dismisses the viewcontroller by swiping down, `dismiss` is not called and the problem in #606 occurs.

By implementing `UIAdaptivePresentationControllerDelegate`, you can receive events of viewcontroller dismissal and perform clean-ups.

> Are both methods required (dismiss and presentationControllerDidDismiss) or is one better to go with than the other?

When viewcontroller is dismissed by `dismissViewControllerAnimated` method, `presentationControllerDidDismiss` is not triggered. So both `dismiss` and `presentationControllerDidDismiss` need to call `clearWebView`.